python/lsst/ts/MTAlignment/mock/mock_t2sa.py

```diff
@@ -0,0 +1,1044 @@
+
+        self._laser_warmup_task: typing.Union[None, asyncio.Task] = None
+        self.laser_warmup_status: asyncio.Future = asyncio.Future()
+
+        self.canned_replies = {
+            "!SET_SIM:0": "ACK300",
+            "!SET_SIM:1": "ACK300",
+            "SET_RANDOMIZE_POINTS:0": "ACK300",
+            "SET_RANDOMIZE_POINTS:1": "ACK300",
+            "!RESET_T2SA": "ACK300",
+            "!NEW_STATION": "ACK300",
+        }
+        self.command_handlers: dict[str, typing.Any] = {
+            "!2FACE_CHECK": self.execute_two_face_check,
+            "!CMDEXE": self.execute_measure_plan,
+            "!HALT": self.halt,
+            "!LOAD_SA_TEMPLATE_FILE": self.load_sa_template_file,
+            "!LST": self.set_power,
+            "!MEAS_DRIFT": self.execute_drift,
+            "!MEAS_SINGLE_POINT": self.measure_single_point,
+            "!PUBLISH_ALT_AZ_ROT": self.set_alt_az_rot,
+            "!SAVE_SA_JOBFILE": self.save_sa_jobfile,
+            "!SET_REFERENCE_GROUP": self.set_reference_group,
+            "!SET_WORKING_FRAME": self.set_working_frame,
+            "?LSTA": self.get_laser_status,
+            "?OFFSET": self.write_target_offset,
+            "?POINT_DELTA": self.measure_point_delta,
+            "?POS CAM": self.write_target_position_cam,
+            "?POS M1M3": self.write_target_position_m1m3,
+            "?POS M2": self.write_target_position_m2,
+            "?STAT": self.write_status,
+        }
+
+        self.command_arg_parsers = {
+            "!2FACE_CHECK": re.compile(r"(?P<target>.*)"),
+            "!CMDEXE": re.compile(r"(?P<target>.*)"),
+            "!LOAD_SA_TEMPLATE_FILE": re.compile(r"(?P<file_path>.*)"),
+            "!LST": re.compile(r"(?P<value>.*)"),
+            "!MEAS_DRIFT": re.compile(r"(?P<target>.*)"),
+            "!MEAS_SINGLE_POINT": re.compile(
+                r"(?P<collection>.*);(?P<point_group>.*);(?P<target_n>.*)"
+            ),
+            "!PUBLISH_ALT_AZ_ROT": re.compile(r"(?P<alt>.*);(?P<az>.*);(?P<rot>.*)"),
+            "!SAVE_SA_JOBFILE": re.compile(r"(?P<filename>.*)"),
+            "!SET_REFERENCE_GROUP": re.compile(r"(?P<reference_group>.*)"),
+            "!SET_WORKING_FRAME": re.compile(r"(?P<working_frame>.*)"),
+            "?OFFSET": re.compile(r"(?P<reference_group>.*);(?P<target>.*)"),
+            "?POINT_DELTA": re.compile(
+                r"(?P<p1collection>.*);(?P<p1group>.*);(?P<p1>.*);"
+                r"(?P<p2collection>.*);(?P<p2group>.*);(?P<p2>.*)"
+            ),
+        }
+
+        self.collection_point_regexp = re.compile(r"(?P<group>.*)_P(?P<index>.*)")
+
+        duplicate_keys = self.command_handlers.keys() & self.canned_replies.keys()
+        if duplicate_keys:
+            raise RuntimeError(
+                f"Bug: keys {duplicate_keys} appear in both "
+                "canned_replies and comamnd_handlers"
+            )
+
+        self._laser_warmup_start: typing.Union[None, float] = None
```

Comment:
Please consider `self._laser_warmup_start_tai`. To me, the current name sounds like a boolean or a method, rather than a timestamp.